Basic_Python_Diploma/VK.py

```diff
@@ -0,0 +1,213 @@
+
+    def user(self, ids=None):
+        if ids:
+            user_params = {"user_ids": ids}
+        else:
+            user_params = {"user_ids": self.id}
+        user_params.update(self.params)
+        if ids:
+            if 0 < len(ids) <= 10:
+                user_id = requests.get(self.API_URL + self.methods['users']['get'],
+                                       params=user_params).json()['response'][0]
+                return user_id
+            else:
+                user_ids = requests.get(self.API_URL + self.methods['users']['get'],
+                                        params=user_params).json()['response']
+                return user_ids
+        else:
+            user_id = requests.get(self.API_URL + self.methods['users']['get'],
+                                   params=user_params).json()['response'][0]
+
+            user_name = user_id['first_name'] + ' ' + user_id['last_name']
+            return user_name
+
+    def mutual_friends(self, friend):
+        mutual_friends_params = {
+            'source_uid': self.id,
+            'target_uid': friend,
+        }
+        mutual_friends_params.update(self.params)
+        ids_list = requests.get(self.API_URL + self.methods['friends']['getMutual'],
+                                params=mutual_friends_params).json()['response']
+
+        ids_str = ''
+        friends_list = [User(_id) for _id in ids_list]
+        for _id in ids_list:
+            if not ids_str:
+                ids_str += str(_id)
+            else:
+                ids_str = ids_str + "," + str(_id)
+        mut_friends_names = self.user(ids_str)
+        mut_friends_names_list = []
+        for _name in mut_friends_names:
+            name = _name['first_name'] + ' ' + _name['last_name']
+            mut_friends_names_list.append(name)
+        print(f'{self.user()} и {User(friend).user()} имеют {len(mut_friends_names_list)} общих друзей:')
+        print(*mut_friends_names_list, sep=", ")
+        print(f'Все они являются сущностями и хранятся в списке "users_list".')
+        print(f'А вот список ссылок на их профили. ')
+        print(*friends_list, sep="\t")
+        print()
+
+    def get_photos(self):
+        param = {"album_id": "profile", "photo_sizes": "w", "type": "z", 'extended': 1, "owner_id": self.id}
+        param.update(self.params)
+        photos_list = requests.get(self.API_URL + self.methods['photos']['get'],
+                                   params=param).json()['response']["items"][-5:]
+        # pprint(photos_list)
+        url_list = [(photo["sizes"][-1]['url'], photo['likes']["count"], photo["date"]) for photo in photos_list]
+        return url_list, self.user()
+
+
+    def download(self, item):
+        target_folder = 'downloads'
+        if not os.path.exists(target_folder):
+            os.mkdir(target_folder)
+        target_folder = os.path.join(target_folder, str(self.id))
+        if not os.path.exists(target_folder):
+            os.mkdir(target_folder)
```

Comment:
Можно сильно упростить эту логику:

```suggestion
        target_folder = os.path.join('downloads', str(self.id))
        os.makedirs(target_folder, exist_ok=True)
```

https://docs.python.org/3/library/os.html#os.makedirs

Reply:
Сделано + метод стал статическим.
Файл VK.py по версии PyCharm полностью соответствует PEP8.